flexx/app/_tornadoserver.py

```diff
@@ -678,13 +678,18 @@ def check_origin(self, origin):
         serving_host = self.request.headers.get("Host")
         serving_hostname = serving_host.split(':')[0]
         connecting_host = urlparse(origin).netloc
+        connecting_hostname = connecting_host.split(':')[0]
         
         if serving_hostname == 'localhost':
             return True  # Safe
         elif serving_hostname == '0.0.0.0':
             return True  # we cannot know if the origin matches
         elif serving_host == connecting_host:
             return True
+        elif serving_hostname == connecting_hostname:
+            # Comparing hostnames in case of Jupyter notebooks
+            # connecting_host and serving_host have same host names but different ports
+            return True
```

Comment:
Maybe the better solution will be in comparison between config.hostname and connecting_hostname?